src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -8938,7 +8938,37 @@ public class MyAttribute : System.Attribute
             comp = CreateCompilation(source, parseOptions: TestOptions.RegularNext);
             comp.VerifyDiagnostics();
 
-            VerifyParameter(comp, 0, "System.Int32 C.this[System.Int32 parameter].get");
+            VerifyParameter(comp, 0, "System.Int32 C.this[System.Int32 parameter] { get; }");
+        }
+
+        [Fact]
+        public void ParameterScope_InMethodAttributeNameOf_Indexer_SetterOnly()
+        {
+            var source = @"
+class C
+{
+    [My(nameof(parameter))]
+    int this[int parameter] { set => throw null; } 
+}
+
+public class MyAttribute : System.Attribute
+{
+    public MyAttribute(string name1) { }
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular10);
+            comp.VerifyDiagnostics(
+                // (4,16): error CS0103: The name 'parameter' does not exist in the current context
+                //     [My(nameof(parameter))]
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "parameter").WithArguments("parameter").WithLocation(4, 16)
```

Comment:
I didn't understand why we give an error here. Is that the intended design?

Reply:
That's the existing C# 10 behavior. 
The new binders/scopes are currently conditional on LangVer preview. The plan is to make that unconditional by the time we ship C# 11, after we've evaluated the compat breaks with preview customers.

Reply:
Oh d'oy. Sorry, I missed the extra CreateCompilation+Verify below.